scripts/vscodeserver/src/vscodeserver.jl

```diff
@@ -180,114 +182,134 @@ function get_modules(toplevel = nothing, mods = Set(Module[]))
     mods
 end
 
-run(conn_endpoint)
-
-@async try
-    while true
-        msg = JSONRPC.get_next_message(conn_endpoint)
-
-        if msg["method"] == "repl/runcode"
-            params = msg["params"]
+const conn_endpoint = Ref{Union{Nothing,JSONRPC.JSONRPCEndpoint}}(nothing)
 
+isactive() = conn_endpoint[] !== nothing
 
-            source_filename = params["filename"]
-            code_line = params["line"]
-            code_column = params["column"]
-            source_code = params["code"]
-            mod = params["module"]
+function serve(args...; is_dev = false)
+    conn = connect(args...)
+    conn_endpoint[] = JSONRPC.JSONRPCEndpoint(conn, conn)
+    run(conn_endpoint[])
 
-            resolved_mod = try
-                module_from_string(mod)
+    if is_dev
```

Comment:
Hm, I'm not so sure I like it that we have a completely different code path at dev. What is the benefit of this?

Reply:
this flag will be automatically set when we start REPL in development mode (which just uses `DEBUG_MODE` flag I set in `launch.json`):
https://github.com/julia-vscode/julia-vscode/blob/42e7402ffda79da310b8fa37d0b86b827b1ff420/scripts/vscodeserver/boot_repl.jl#L40

The main benefit here is as I mentioned:
> REPL will keep living even when an internal error happens when in extension development

In other word, while developing interactive code, we often have internal error, which currently causes this `@async` block just to be killed, and then we need to reboot REPL.

Another benefit is to use `Base.invokelatest` so that we can have every hot reload of message handlers.

---

Well, this is not exactly related to "packaging up", so I'm okay to do this in another PR if you want.

Reply:
Ok, sounds good, lets leave it like you changed it!